sumpy/expansion/diff_op.py

```diff
@@ -157,25 +252,52 @@ def gradient(diff_op):
     eqs = []
     dim = diff_op.dim
     for i in range(dim):
-        mi = [0]*dim
+        mi = [0]*diff_op.total_dims
         mi[i] = 1
         eqs.append(diff(diff_op, tuple(mi)).eqs[0])
     return LinearPDESystemOperator(dim, *eqs)
 
 
-def concat(op1, op2):
-    assert op1.dim == op2.dim
-    eqs = list(op1.eqs)
-    eqs.extend(list(op2.eqs))
-    return LinearPDESystemOperator(op1.dim, *eqs)
+def curl(pde):
+    assert len(pde.eqs) == 3
+    assert pde.dim == 3
+    eqs = []
+    mis = []
+    for i in range(3):
+        mi = [0]*pde.total_dims
+        mi[i] = 1
+        mis.append(tuple(mi))
+
+    for i in range(3):
+        new_pde = diff(pde[(i+2) % 3], mis[(i+1) % 3]) - \
+            diff(pde[(i+1) % 3], mis[(i+2) % 3])
+        eqs.append(new_pde.eqs[0])
+
+    return LinearPDESystemOperator(pde.dim, *eqs)
+
+
+def concat(*ops):
+    ops = list(ops)
+    assert len(ops) >= 1
+    dim = ops[0].dim
+    for op in ops:
+        assert op.dim == dim
+    eqs = list(ops[0].eqs)
+    for op in ops[1:]:
+        eqs.extend(list(op.eqs))
+    return LinearPDESystemOperator(dim, *eqs)
 
 
-def make_identity_diff_op(ninput, noutput=1):
+def make_identity_diff_op(ninput, noutput=1, include_time=False):
     """
     Returns the identity as a linear PDE system operator.
-    :arg ninput: number of input variables to the function
+    :arg ninput: number of spatial variables of the function
     :arg noutput: number of output values of function
+    :arg include_time: include time as a dimension
     """
-    mi = tuple([0]*ninput)
+    if include_time:
+        mi = tuple([0]*(ninput + 1))
+    else:
+        mi = tuple([0]*ninput)
```

Comment:
Why is time being handled specially here?

Reply:
Issue was that when we say `laplacian(u)` it means laplacian with spatial variables. We don't want to do `d^2/dx^2 + d^2/dy^2 + d^2/dt^2`. I agree that it's not pretty, and I'm open to suggestions.

Reply:
I guess just document that there's a convention that the first `dim` partial derivatives in the PDE operator refer to the spatial variables.